### PR TITLE
Kan 15 ibrahim mostafa tc 8

### PR DESCRIPTION
**Pull request by ibrMostafa333**

### CODE REVIEW

File: .idea/misc.xml
Comment (Omarhani) [inline .idea/misc.xml:1]: remove this file from your commit

File: SignUpTests.xml [hunk 1]
```diff
@@ -1,11 +1,10 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <!DOCTYPE suite SYSTEM "http://testng.org/testng-1.0.dtd">
-<suite name="regression test suite" >
-    <test name="Chrome" >
+<suite name="regression test suite">
```
Comment (Omarhani) [inline SignUpTests.xml:3]: remove this file from your commit

File: data/testData.json [hunk 2]
```diff
@@ -1,21 +1,20 @@
 {
-  "URL" : "https://automationexercise.com/",
+  "URL": "https://automationexercise.com/",
```
Comment (Omarhani) [inline data/testData.json:2]: remove this file from your commit

File: data/testData.json [hunk 3]
```diff
@@ -1,21 +1,20 @@
...
   "Login": {
     "ValidCredentials": {
-      "Email": "eman@gmail.com",
-      "Password": "eman"
+      "Username": "",
```
Comment (Omarhani) [inline data/testData.json:5]: keep this credentials

File: report/report.html
Comment (Omarhani) [inline report/report.html:1]: remove this file from your commit

File: smoke.xml
Comment (Omarhani) [inline smoke.xml:1]: out off scope

File: src/main/java/pages/DeleteAccountPage.java [hunk 4]
```diff
@@ -9,7 +9,7 @@ public DeleteAccountPage(WebDriver driver) {
         super(driver);
     }
 
-    private final By deleteAccountText = By.xpath("//b[contains(text(),'Account Deleted!')]");
+    final static By deleteAccountText = By.xpath("//b[contains(text(),'Account Deleted!')]");
```
Comment (Omarhani) [inline src/main/java/pages/DeleteAccountPage.java:12]: private final is correct

File: src/main/java/pages/LoginPage.java
Comment (Omarhani) [inline src/main/java/pages/LoginPage.java:1]: remove this changes

File: src/main/java/pages/SignUpPage.java [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+    public void enterEmail(String email) {
+        sendKeys(emailField,5,email);
+    }
+
+    public void clickSignupBtn() {
+        click(SignupBtn,5);
+    }
+
+    public void ExistSignUpFuc(String name,String email){
```
Comment (Omarhani) [inline src/main/java/pages/SignUpPage.java:36]: camel format
signUpFeature()

File: src/main/java/pages/SignUpPage.java [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+        enterName(name);
+        enterEmail(email);
+        clickSignupBtn();
+    }
+
+    public String ExistSignUpValidation() {
+        return driver.findElement(ExistSignUpMessage).getText();
```
Comment (Omarhani) [inline src/main/java/pages/SignUpPage.java:43]: user getText() function from MethodHandles

File: src/test/java/base/BaseTests.java [hunk 7]
```diff
@@ -33,15 +33,15 @@ public class BaseTests {
 
     @Parameters("browser")
     @BeforeClass
-    public void setUp(@Optional("chrome") String browser) {
+    public void setUp(@Optional("Chrome") String browser) {
```
Comment (Omarhani) [inline src/test/java/base/BaseTests.java:36]: keep chrome with c

File: src/test/java/data/DataModel.java [hunk 8]
```diff
@@ -2,8 +2,6 @@
 
 public class DataModel {
     public String URL;
-
-    public Login Login;
-
-    public Object[][] Login1;
+    public Register register;
```
Comment (Omarhani) [inline src/test/java/data/DataModel.java:5]: Register Register

File: src/test/java/data/DataModel.java [hunk 9]
```diff
@@ -2,8 +2,6 @@
 
 public class DataModel {
     public String URL;
-
-    public Login Login;
```
Comment (Omarhani) [inline src/test/java/data/DataModel.java:6]: keep Login

File: src/test/java/data/ValidCredentials.java [hunk 10]
```diff
@@ -2,6 +2,6 @@
 
 public class ValidCredentials {
 
-    public String Email;
```
Comment (Omarhani) [inline src/test/java/data/ValidCredentials.java:5]: roll back

File: src/test/java/homePage_subscription/HomePageSubscriptionTest.java [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+import org.testng.annotations.Test;
+
+public class HomePageSubscriptionTest extends BaseTests {
+
+    @Test
+    public void SubscriptionTest() {
+        Assert.assertTrue(homePage.homePageTitle().contains("Automation Exercise"));
+        homePage.ScrollToSubscription();
+        Assert.assertTrue(homePage.VerifySubscribDisplay().contains("SUBSCRIPTION"));
+        homePage.EnterSubscribeEmail(dataModel.register.Email);
```
Comment (Omarhani) [inline src/test/java/homePage_subscription/HomePageSubscriptionTest.java:14]: camel format

File: src/test/java/login/LoginTests.java
Comment (Omarhani) [inline src/test/java/login/LoginTests.java:1]: Roll Back

File: src/test/java/products/ProductsTests.java [hunk 12]
```diff
@@ -0,0 +1,19 @@
...
+import pages.ProductsPage;
+
+public class ProductsTests extends BaseTests {
+
+    @Test
+    public void ProductsInformationTest() {
+
+        Assert.assertTrue(homePage.homePageTitle().contains("Automation Exercise"));
+        ProductsPage productsPage = homePage.clickProductsLink();
+        Assert.assertTrue(productsPage.getAllProductPageTitle().contains("All Products"));
```
Comment (Omarhani) [inline src/test/java/products/ProductsTests.java:15]: remove hard-coded data

File: src/test/java/signUp/SignUpTests.java
Comment (Omarhani) [inline src/test/java/signUp/SignUpTests.java:1]: Roll back

File: data/testData.json (hunk 3 above)
Comment (ibrMostafa333) [reply data/testData.json:5]: Done

File: data/testData.json (hunk 2 above)
Comment (ibrMostafa333) [reply data/testData.json:2]: Done

File: src/test/java/homePage_subscription/HomePageSubscriptionTest.java (hunk 11 above)
Comment (ibrMostafa333) [reply src/test/java/homePage_subscription/HomePageSubscriptionTest.java:14]: Done

File: src/test/java/login/LoginTests.java
Comment (ibrMostafa333) [reply src/test/java/login/LoginTests.java:1]: Done

**Pull request closed without merging**